nocasedict/_nocasedict.py

```diff
@@ -0,0 +1,475 @@
+#
+# (C) Copyright 2003-2007 Hewlett-Packard Development Company, L.P.
+# (C) Copyright 2006-2007 Novell, Inc.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+# Author: Tim Potter <tpot@hp.com>
+# Author: Martin Pool <mbp@hp.com>
+# Author: Bart Whiteley <bwhiteley@suse.de>
+#
+
+"""
+The only class exposed by this package is :class:`nocasedict.NocaseDict`.
+"""
+
+from __future__ import print_function, absolute_import
+
+import sys
+import warnings
+from collections import OrderedDict
+
+import six
+
+__all__ = ['NocaseDict']
+
+
+class NocaseDict(object):
+    """
+    Yet another implementation of a case-insensitive dictionary.
+
+    Whenever keys are looked up, that is done case-insensitively. Whenever
```

Comment:
How about "Key lookup is always case insensitive."

Reply:
Has already been improved in docs update PR, please review there again.